### ISSUE TITLE
`@override` is not detected when re-exported

### ISSUE THREAD

**Issue by bersbersbers** [astral-sh/ruff#5400]
https://github.com/astral-sh/ruff/issues/3910 is great! However, it does not work for me when re-exporting `override`:

`override.py`
```python
__all__ = ["override"]
from typing_extensions import override
```

`bug.py`
```python
from override import override

@override
def BAD_FUN() -> None:
    pass
```

* `ruff bug.py --select N802 --isolated`
* Python 3.11.4, ruff 0.0.275

Output:
```
bug.py:5:5: N802 Function name `BAD_FUN` should be lowercase
```

**Comment by charliermarsh** [comment]
Can you try marking your `override` module via [`typing-modules`](https://beta.ruff.rs/docs/settings/#typing-modules)?
